node/external/transactionAPI/gasUsedAndFeeProcessor.go

```diff
@@ -27,11 +45,37 @@ func (gfp *gasUsedAndFeeProcessor) computeAndAttachGasUsedAndFee(tx *transaction
 	tx.GasUsed = gasUsed
 	tx.Fee = fee.String()
 
-	if tx.IsRelayed || gfp.isESDTOperationWithSCCall(tx) {
+	isFeeFixActive := gfp.enableEpochsHandler.IsFlagEnabledInEpoch(common.FixRelayedBaseCostFlag, tx.Epoch)
+	isRelayedBeforeFix := tx.IsRelayed && !isFeeFixActive
+	if isRelayedBeforeFix || gfp.isESDTOperationWithSCCall(tx) {
 		tx.GasUsed = tx.GasLimit
 		tx.Fee = tx.InitiallyPaidFee
 	}
 
+	// if there is a guardian operation, SetGuardian/GuardAccount/UnGuardAccount
+	// the pre-configured cost of the operation must be added separately
+	if gfp.isGuardianOperation(tx) && isFeeFixActive {
+		gasUsed = gfp.feeComputer.ComputeGasLimit(tx)
+		guardianOperationCost := gfp.getGuardianOperationCost(tx)
+		gasUsed += guardianOperationCost
```

Comment:
Not a request change: having this logic here, it makes one think that it also resides in other places (since here we are in a mere API component). In the future, we might forget to adjust it (in API), if the actual processing flow / logic needs to be adjusted.